scripts/o3de/o3de/manifest.py

```diff
@@ -357,7 +358,17 @@ def get_gem_external_subdirectories(gem_path: pathlib.Path, visited_gem_paths: l
         return []
     visited_gem_paths.append(gem_path)
 
-    gem_object = get_gem_json_data(gem_path=gem_path)
+    if isinstance(gems_json_data_by_path, dict):
+        # Use the cache 
+        if gem_path in gems_json_data_by_path:
+            gem_object = gems_json_data_by_path[gem_path]
+        else:
+            gem_object = get_gem_json_data(gem_path=gem_path)
+            # store the value even if its None so we don't open the file again
+            gems_json_data_by_path[gem_path] = gem_object
+    else:
+        gem_object = get_gem_json_data(gem_path=gem_path)
+
```

Comment:
Thanks for adding the caching mechanism to the gem's external subdirectories.

Reply:
Good idea.  At first I just tried setting the default arg of `get_gem_external_subdirectories` to be `dict()` instead of `None` but that caused the tests to fail because apparently that data is persisted across calls to the top level function which was unexpected and could be a source of subtle bugs in Project Manager, so I resorted to passing an empty `dict()` object to the function at the call sites.

Reply:
Yeah, that is because python is more of a pass by name paradigm.
So setting a default value of `gems_json_data_by_path = {}` binds a reference to the dictionary of `{}` to `gems_json_data_by_path` that gets used every time the function is invoked without a parameter